### PR TITLE
Various bugfixes

### PR DESCRIPTION
**Pull request by smathot**
@dschreij I will append various minor bugfix commits to this PR, for you to review when you get back :-)

### CODE REVIEW

File: src/js/osweb/backends/canvas.js [hunk 1]
```diff
@@ -893,6 +893,9 @@ export default class Canvas {
    * @param {Object} styleArgs - Optional styling argument for the element.
    */
   text (txt, center, x, y, html, styleArgs) {
+    // Newlines are not dealt with correctly, so we convert them to <br />
+    // tags, which results in the same behavior as the desktop.
+    txt = txt.replace('\n', '<br />')
```
Comment (dschreij) [inline src/js/osweb/backends/canvas.js:898]: The space between r and / is not strictly required, but also not a problem to leave in.

File: src/js/osweb/items/loop.js [hunk 2]
```diff
@@ -79,15 +78,14 @@ export default class Loop extends Item {
 
   /** Reset all item variables to their default value. */
   reset () {
-    this.matrix = []
+    this.orig_matrix = []
```
Comment (dschreij) [inline src/js/osweb/items/loop.js:81]: It is really hard to follow what you are doing here, or why the orig_matrix is required to make resumptions possible, but I assume you tested it and this works. If so, I am ok with it.

File: src/js/osweb/items/loop.js [hunk 3]
```diff
@@ -134,52 +132,41 @@ export default class Loop extends Item {
...
-              this.matrix[cycle] = {}
+            if (this.orig_matrix[cycle] === undefined) {
+              this.orig_matrix[cycle] = {}
             }
-            this.matrix[cycle][name] = value
+            this.orig_matrix[cycle][name] = value
             break
           case 'fullfactorial':
             this._operations.push([fullfactorial, []])
-            // this.matrix = fullfactorial(this.matrix)
```
Comment (dschreij) [inline src/js/osweb/items/loop.js:144]: Oops, I guess I forgot to clean up after myself again ;)

File: src/js/osweb/classes/syntax.js [hunk 4]
```diff
@@ -153,18 +148,27 @@ export default class Syntax {
...
-    if (result !== '') {
-      const nr = toNumber(result)
+    if (text !== '') {
+      const nr = toNumber(text)
       if (!isNaN(nr)) return nr
     }
     // Check if content has additional quotes
-    return this.strip_slashes(this.unescapeBrackets(result))
+    return this.strip_slashes(this.unescapeBrackets(text))
   }
```
Comment (dschreij) [inline src/js/osweb/classes/syntax.js:173]: Nice! This is also easier to follow this way

**Review by dschreij — APPROVED** [review]

File: src/js/osweb/items/loop.js (hunk 2 above)
Comment (smathot) [reply src/js/osweb/items/loop.js:81]: The `orig_matrix` is the original state of the matrix before any of the operations are applied. And then the `matrix` is the matrix as it used during a particular run of the `loop` item. This allows the loop item to run multiple times from the same state.

**Comment by smathot** [comment]
Sorry, I didn't notice that you had approved the PR so I appended c2b84c35c6da332c6e9a69460c6388e17b057b01. Could you quickly review that one too? And then if you approve I suggest that we release 1.3.9, and at the same time release OpenSesame 3.3.4 with *many* bugfixes!

File: src/js/osweb/classes/syntax.js [hunk 5]
```diff
@@ -222,6 +222,39 @@ export default class Syntax {
...
+   * single tab.
+   * @param {String} script - A script
+   * @return {String} - A dedented script
+   */
+  dedent (script) {
+    const lines = script.split('\n')
+    let dedented = []
+    for (const line of lines) {
+      if (line[0] !== '\t') {
+        return script
```
Comment (dschreij) [inline src/js/osweb/classes/syntax.js:237]: This seems peculiar. If the first character of a line is not a tab, return the script (e.g. the input variable?) so for instance a script like
```
    define something
    do something else
non-indented line
```
would just return the script since the last line does not start with a tab, even though you might still want to dedent the first two lines?
I assume this is just my lack of understanding the situation and you have tested this (judging from the unit tests) so it just works.

File: src/js/osweb/items/inline_script.js [hunk 6]
```diff
@@ -56,54 +56,6 @@ export default class InlineScript extends Item {
     this.vars._run = ''
   }
 
-  /**
```
Comment (dschreij) [inline src/js/osweb/items/inline_script.js:59]: inline_script is not supported anyway, right? Should we leave this file in at all?

**Review by dschreij — APPROVED** [review]

File: src/js/osweb/classes/syntax.js (hunk 5 above)
Comment (smathot) [reply src/js/osweb/classes/syntax.js:237]: Yes, that's intended. Basically, it should dedent a script that corresponds to an item definition, and this is always indented by a single tab. If any line is not indented, then it's not an item definition, and it should be left as is.
